File: 0013_Roman_to_Integer/solution.cpp
#include <iostream>
#include <unordered_map>
#include <string>
using namespace std;
class Solution {
    public:
        int romanToInt(string s) {
            unordered_map<char, int> roman = {
                {'I', 1},
                {'V', 5},
                {'X', 10},
                {'L', 50},
                {'C', 100},
                {'D', 500},
                {'M', 1000}
            };
        int num=0;
        int prev=0;
        for(int i=s.length()-1; i>=0; i--){
            int curr=roman[s[i]];
            if(curr<prev){
                num=num-curr;
            } else{
                num=num+curr;
            }
            prev=curr;
        }
        return num;
    }
};
int main() {
    Solution sol;
    string data;
    cout << "Enter a Roman numeral: ";
    cin >> data;
    int value = sol.romanToInt(data);
    cout << "The integer value is: " << value << endl;
    return 0;
}

File: 0013_Roman_to_Integer/solution.java
import java.util.Scanner;
import java.util.HashMap;

public class Solution{
    public int romanToInt(String s){
        HashMap<Character, Integer> roman = new HashMap<>();
        roman.put('I', 1);
        roman.put('V', 5);
        roman.put('X', 10);
        roman.put('L', 50);
        roman.put('C', 100);
        roman.put('D', 500);
        roman.put('M', 1000);

        int num = 0;
        int prev = 0;
        for(int i = s.length()-1; i>=0; i--){
            char c = s.charAt(i);
            int curr = roman.get(c);
            if (curr < prev){
                num= num-curr;
            } else {
                num = num+curr;
            }
            prev = curr;
        }
        return num;
    }
}
public class Main{
    public static void main(String[] args){
        Solution sol = new Solution();
        string data;
        System.out.println("Enter a Roman numeral:")
        Scanner sc = new Scanner(System.in);
        data = sc.nextLine();
        int value = sol.romanToInt(data);
        System.out.println("The integer value of is: " + value);
        sc.close();
    }
}